controllers/managedocs_controller_test.go

```diff
@@ -841,6 +847,56 @@ var _ = Describe("ManagedOCS controller", func() {
 				}, timeout, interval).Should(Equal(true))
 			})
 		})
+		When("the OCS CSV resource is created", func() {
+			It("should patch the OCS CSV to set resources for required pods", func() {
+				ocsCSV := ocsCSVTemplate.DeepCopy()
+				ocsCSVInitKey := utils.GetResourceKey(ocsCSV)
+				Expect(k8sClient.Get(ctx, ocsCSVInitKey, ocsCSV)).Should(Succeed())
+
+				deployments := ocsCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+				for i := range deployments {
+					containers := deployments[i].Spec.Template.Spec.Containers
+					for j := range containers {
+						switch container := &containers[j]; container.Name {
+						case "ocs-operator":
+							Expect(container.Resources).Should(Equal(ctrlutils.GetResourceRequirements("ocs-operator")))
+						case "rook-ceph-operator":
+							Expect(container.Resources).Should(Equal(ctrlutils.GetResourceRequirements("rook-ceph-operator")))
+						case "ocs-metrics-exporter":
+							Expect(container.Resources).Should(Equal(ctrlutils.GetResourceRequirements("ocs-metrics-exporter")))
+						}
+					}
+				}
+			})
+		})
+		When("the OCS CSV resource is modified", func() {
+			It("should revert the changes in OCS CSV to have provided resource requirements", func() {
+				ocsCSV := ocsCSVTemplate.DeepCopy()
+				ocsCSVInitKey := utils.GetResourceKey(ocsCSV)
+				Expect(k8sClient.Get(ctx, ocsCSVInitKey, ocsCSV)).Should(Succeed())
+
+				deployments := ocsCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+				for i := range deployments {
+					containers := deployments[i].Spec.Template.Spec.Containers
+					for j := range containers {
+						switch container := &containers[j]; container.Name {
+						case "ocs-operator":
+							container.Resources.Limits = corev1.ResourceList{
+								"cpu":    resource.MustParse("3000m"),
+								"memory": resource.MustParse("8Gi"),
+							}
+							Expect(k8sClient.Update(ctx, ocsCSV)).Should(Succeed())
+
+							// Wait for the spec changes to be reverted
+							Eventually(func() corev1.ResourceRequirements {
+								Expect(k8sClient.Get(ctx, ocsCSVInitKey, ocsCSV)).Should(Succeed())
+								return ocsCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec.Template.Spec.Containers[j].Resources
+							}, timeout, interval).Should(Equal(ctrlutils.GetResourceRequirements("ocs-operator")))
```

Comment:
Move the expected test outside of your mutation code. (outside the loop), the code structure should be:

1. prepare - loading the CSV
2. create test condition - the loop that finds the container and mutates it
3. wait for the result and assert - the `Eventually`